### PR TITLE
Refactor devices controller

### PR DESCRIPTION
**Pull request by t-harps**
Moved hash method to model and made necessary changes

### CODE REVIEW

File: app/models/device.rb [hunk 1]
```diff
@@ -43,4 +43,10 @@ def change_privilege_for(dev, new_privilege)
     record.privilege = new_privilege
     record.save
   end
+
+  def device_checkin_hash
+    hash = self.as_json
+    hash[:last_checkin] = self.checkins.last.get_data if self.checkins.exists?
```
Comment (dansteele) [inline app/models/device.rb]: Do you need to call 'self' here?
@t-harps

Comment (t-harps) [inline app/models/device.rb]: What could we use instead?

Comment (dansteele) [inline app/models/device.rb]: @t-harps self is the first thing that's called in the call stack
Try just taking it out and seeing if it still works :)

Comment (t-harps) [inline app/models/device.rb]: Fair enough, still works! will recommit.

**Pull request merged**